src/Evolution/DiscontinuousGalerkin/SlopeLimiters/WenoGridHelpers.hpp

```diff
@@ -0,0 +1,57 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+/// \cond
+class DataVector;
+template <size_t>
+class Direction;
+template <size_t>
+class Mesh;
+/// \endcond
+
+namespace SlopeLimiters {
+namespace Weno_detail {
+
+// Compute the coordinate positions of a neighbor element's grid points in the
+// logical coordinates of the local element.
+//
+// However, the results are organized to match the expected input of a
+// RegularGridInterpolator constructor:
+// - The coordinates are given as an array of 1D coordinate values.
+// - For any dimension where the neighbor and local element share the same 1D
+//   mesh, the array is empty. (This tells the RegularGridInterpolator that no
+//   interpolation needs to be done in this dimension.)
+//
+// NOTE: Current implementation assumes, but does not check:
```

Comment:
Why not add these as an ASSERT?

Reply:
- I could pass in the local `Element` as a new argument, used _only_ to get count its neighbors for the h-ref ASSERT. Should I do this?
- I could pass in the neighbor meshes, as obtained from the limiter communicated data, to check for p-ref. However this is cumbersome, as it requires either templating this function on the general data package type, or repacking the neighbor mesh data into something of knowable type... Should I do this?

Reply:
On second thought: I probably should do these, because these arguments will be necessary in the future anyway when generalizing this function to h/p-refinement.